packages/cli/src/lib/inspect/setup.ts

```diff
@@ -29,17 +44,37 @@ export const setupCliWorld = async (
   setJsonMode(Boolean(flags.json));
   setVerboseMode(Boolean(flags.verbose));
 
+  // Check for updates
+  const cacheFile = join(getWorkflowCacheDir(), 'version-check.json');
+  const updateCheck = await checkForUpdateCached(version, cacheFile);
+
   const withAnsiLinks = flags.json ? false : true;
   const docsUrl = withAnsiLinks
     ? terminalLink('https://useworkflow.dev/', 'https://useworkflow.dev/')
     : 'https://useworkflow.dev/';
 
-  logger.showBox(
-    'green',
+  // Prepare showBox lines
+  const boxLines = [
     `Workflow CLI v${version}`,
     `Docs at ${docsUrl}`,
-    chalk.yellow('This is a beta release')
-  );
+    chalk.yellow('This is a beta release'),
+  ];
+
+  // Add update message if available
+  if (updateCheck.needsUpdate && updateCheck.latestVersion) {
+    boxLines.push(
+      '',
+      chalk.cyan(
+        `Update available: ${updateCheck.currentVersion} → ${updateCheck.latestVersion}`
+      ),
+      // Note that we're suggesting install "latest" instead of the release tag that the user is
+      // on, because we currently tag beta releases as "latest". After GA, we need to adjust
+      // this to install the release tag that the user is on.
+      chalk.gray('Run: `[npm|yarn|pnpm] i workflow@latest`')
```

Comment:
no `bun`? 😢 

`bun` > `yarn` :)

Reply:
also if you install `@latest` I think it will use `@latest` in the user's package.json - so I'd actually recommend `pnpm i workflow@${updateCheck.latestVersion}` I think